api/incubator/src/main/java/io/opentelemetry/api/incubator/config/ConfigProvider.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.incubator.config;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A registry for accessing declarative configuration.
+ *
+ * <p>The name <i>Provider</i> is for consistency with other languages and it is <b>NOT</b> loaded
+ * using reflection.
+ *
+ * <p>See {@link InstrumentationConfigUtil} for convenience methods for extracting config from
+ * {@link ConfigProvider}.
+ */
+@ThreadSafe
+public interface ConfigProvider {
```

Comment:
The PR description seems to focus on the instrumentation configuration. Should the name be more specific? For example  `InstrumentationConfigProvider`?